CMakePresets.json

```diff
@@ -167,6 +167,22 @@
         "VCPKG_CHAINLOAD_TOOLCHAIN_FILE": "${sourceDir}/toolchains/Linux-arm.cmake"
       }
     },
+    {
+      "name": "ci-freebsd-x64",
```

Comment:
you should also add CI build/release jobs for it

this preset and toolchain are not needed in this project otherwise

Reply:
You're saying this isn't the right place for the preset if we don't plan on shipping a C-SDK release artifact for FreeBSD. Would the preset exist only in documentation?

I'm unsure whether it's valuable to ship a release at this point, but it'd be good to have a test for it nonetheless so we'll know if we're drifting away from FreeBSD compatibility.

Up to this point, I've used the present only to build on FreeBSD, not to cross-build for FreeBSD. To test cross-building in GitHub, I added a matrix iteration for this preset, but it failed for reasons that aren't obvious from a scan of the stdout/err.

I was unable to add to this branch at all due to GitHub branch protection preventing me from pushing unsigned commits from other authors that I acquired in my branch while rebasing on main. Instead, I pushed to a temporary branch: https://github.com/openziti/ziti-sdk-c/actions/runs/8396544604/job/22998135638

Reply:
I would prefer to add this preset to our CI and publish, that will enable FreeBSD support for downstream SDKs (python, nodejs, etc)

you should be able to rebase your branch against main and push

Reply:
OK, I'll work on getting the FreeBSD target working in GitHub. Do you have any insight based on the failed run I linked?

I'm still unable to force push after rebasing because the main branch still has unverified commits that are pulled in during the rebase, which breaches the branch protection rules. Here the shell transcript showing the rebase and failed push.

```bash
❯ git rebase -i origin/main
Auto-merging library/utils.c
Auto-merging tests/zitilib-tests.cpp
Auto-merging toolchains/FreeBSD-x64.cmake
CONFLICT (add/add): Merge conflict in toolchains/FreeBSD-x64.cmake
error: could not apply 6ac9b96... port to freebsd
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 6ac9b96... port to freebsd

ziti-sdk-c (REBASING 4/5) *​18s 
❯ git rebase --continue
[detached HEAD 72ce6ce] port to freebsd
 Date: Tue Feb 20 17:40:01 2024 -0500
 7 files changed, 41 insertions(+), 2 deletions(-)
 create mode 100644 toolchains/FreeBSD-x64.cmake
Successfully rebased and updated refs/heads/freebsd.

ziti-sdk-c freebsd
❯ g
On branch freebsd
nothing to commit, working tree clean

ziti-sdk-c freebsd
❯ git push -f
Enumerating objects: 28, done.
Counting objects: 100% (28/28), done.
Delta compression using up to 16 threads
Compressing objects: 100% (15/15), done.
Writing objects: 100% (15/15), 2.10 KiB | 2.10 MiB/s, done.
Total 15 (delta 13), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (13/13), completed with 13 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/freebsd.
remote: error: Commits must have verified signatures.
To github.com:openziti/ziti-sdk-c.git
 ! [remote rejected] freebsd -> freebsd (protected branch hook declined)
error: failed to push some refs to 'github.com:openziti/ziti-sdk-c.git'
```

Reply:
The toolchain file needs some tweaks.

Here's the only line containing "error" in the attached log that's mentioned in the vcpkg output. 

```
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv_cpumask_size’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1527:10: error: ‘CPU_SETSIZE’ undeclared (first use in this function); did you mean ‘__CPU_SETSIZE’?
```

I ran the latest vcpkg release on the `freebsd` branch to configure cmake with this command:

```bash
cmake \
  --preset ci-freebsd-x64 \
  -B ./build ; 
```

```
Detecting compiler hash for triplet x64-linux...
Compiler found: /usr/bin/c++
Detecting compiler hash for triplet x64-freebsd...
Compiler found: /usr/bin/c++
The following packages will be built and installed:
    catch2:x64-freebsd@3.4.0 -- /usr/local/vcpkg/buildtrees/versioning_/versions/catch2/5796c1c0513a7b49f135e8acdd1976f53e9944ea
    libuv:x64-freebsd@1.46.0#1 -- /usr/local/vcpkg/buildtrees/versioning_/versions/libuv/2267bb7c14ae2201b02e2d708631e4a228355543
    llhttp:x64-freebsd@9.1.3 -- /usr/local/vcpkg/buildtrees/versioning_/versions/llhttp/3422384dc0ec7e58827c138826bb666efe850eaa
    openssl:x64-freebsd@3.1.4#1 -- /usr/local/vcpkg/buildtrees/versioning_/versions/openssl/5ee74387a1dd212dc9754c614809ff86bd0638fa
  * protobuf:x64-linux@3.21.12 -- /usr/local/vcpkg/buildtrees/versioning_/versions/protobuf/3b145508ba614fe26989b23f6317f15bf6467be4
  * protobuf:x64-freebsd@3.21.12 -- /usr/local/vcpkg/buildtrees/versioning_/versions/protobuf/3b145508ba614fe26989b23f6317f15bf6467be4
    protobuf-c:x64-freebsd@1.4.1#1 -- /usr/local/vcpkg/buildtrees/versioning_/versions/protobuf-c/32d120b5b3a9225b973b135f632ee82be3e7860b
  * vcpkg-cmake-get-vars:x64-linux@2023-03-02 -- /usr/local/vcpkg/buildtrees/versioning_/versions/vcpkg-cmake-get-vars/2e624c2cf12a97a7a802e31ff1d28b9fa6ba9bde
    zlib:x64-freebsd@1.3 -- /usr/local/vcpkg/buildtrees/versioning_/versions/zlib/5ac18c6e6e3e2bf5a9e3d0bc8a845f198e4c4e05
Additional packages (*) will be modified to complete this operation.
Restored 0 package(s) from /.cache/vcpkg/archives in 13.4 us. Use --debug to see more details.
Installing 1/9 libuv:x64-freebsd@1.46.0#1...
Building libuv:x64-freebsd@1.46.0#1...
/usr/local/vcpkg/triplets/community/x64-freebsd.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
/usr/local/vcpkg/buildtrees/versioning_/versions/libuv/2267bb7c14ae2201b02e2d708631e4a228355543: info: installing overlay port from here
-- Downloading https://github.com/libuv/libuv/archive/v1.46.0.tar.gz -> libuv-libuv-v1.46.0.tar.gz...
-- Extracting source /usr/local/vcpkg/downloads/libuv-libuv-v1.46.0.tar.gz
-- Applying patch fix-build-type.patch
-- Using source at /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean
-- Configuring x64-freebsd
-- Building x64-freebsd-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/local/bin/cmake --build . --config Debug --target install -- -v -j17
    Working Directory: /usr/local/vcpkg/buildtrees/libuv/x64-freebsd-dbg
    See logs for more information:
      /usr/local/vcpkg/buildtrees/libuv/install-x64-freebsd-dbg-out.log

Call Stack (most recent call first):
  /github/workspace/build/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  /github/workspace/build/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  buildtrees/versioning_/versions/libuv/2267bb7c14ae2201b02e2d708631e4a228355543/portfile.cmake:23 (vcpkg_cmake_install)
  scripts/ports.cmake:175 (include)


error: building libuv:x64-freebsd failed with: BUILD_FAILED
Elapsed time to handle libuv:x64-freebsd: 3 s
Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+libuv
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?title=[libuv]+Build+error+on+x64-freebsd&body=Copy+issue+body+from+%2Fgithub%2Fworkspace%2Fbuild%2Fvcpkg_installed%2Fvcpkg%2Fissue_body.md
```

Full `/usr/local/vcpkg/buildtrees/libuv/install-x64-freebsd-dbg-out.log`:

```
[1/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/strtok.c.o -MF CMakeFiles/uv_a.dir/src/strtok.c.o.d -o CMakeFiles/uv_a.dir/src/strtok.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/strtok.c
[2/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/strscpy.c.o -MF CMakeFiles/uv_a.dir/src/strscpy.c.o.d -o CMakeFiles/uv_a.dir/src/strscpy.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/strscpy.c
[3/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/version.c.o -MF CMakeFiles/uv_a.dir/src/version.c.o.d -o CMakeFiles/uv_a.dir/src/version.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/version.c
[4/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/thread-common.c.o -MF CMakeFiles/uv_a.dir/src/thread-common.c.o.d -o CMakeFiles/uv_a.dir/src/thread-common.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/thread-common.c
[5/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/uv-data-getter-setters.c.o -MF CMakeFiles/uv_a.dir/src/uv-data-getter-setters.c.o.d -o CMakeFiles/uv_a.dir/src/uv-data-getter-setters.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/uv-data-getter-setters.c
[6/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/dl.c.o -MF CMakeFiles/uv_a.dir/src/unix/dl.c.o.d -o CMakeFiles/uv_a.dir/src/unix/dl.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/dl.c
[7/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/idna.c.o -MF CMakeFiles/uv_a.dir/src/idna.c.o.d -o CMakeFiles/uv_a.dir/src/idna.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/idna.c
[8/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/random.c.o -MF CMakeFiles/uv_a.dir/src/random.c.o.d -o CMakeFiles/uv_a.dir/src/random.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/random.c
[9/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/core.c.o -MF CMakeFiles/uv_a.dir/src/unix/core.c.o.d -o CMakeFiles/uv_a.dir/src/unix/core.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c
FAILED: CMakeFiles/uv_a.dir/src/unix/core.c.o 
/usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/core.c.o -MF CMakeFiles/uv_a.dir/src/unix/core.c.o.d -o CMakeFiles/uv_a.dir/src/unix/core.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv__accept’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:93:22: warning: implicit declaration of function ‘accept4’; did you mean ‘accept’? [-Wimplicit-function-declaration]
 # define uv__accept4 accept4
                      ^
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:555:14: note: in expansion of macro ‘uv__accept4’
     peerfd = uv__accept4(sockfd, NULL, NULL, SOCK_NONBLOCK|SOCK_CLOEXEC);
              ^~~~~~~~~~~
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv__dup2_cloexec’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1091:7: warning: implicit declaration of function ‘dup3’; did you mean ‘dup2’? [-Wimplicit-function-declaration]
   r = dup3(oldfd, newfd, O_CLOEXEC);
       ^~~~
       dup2
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv_cpumask_size’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1527:10: error: ‘CPU_SETSIZE’ undeclared (first use in this function); did you mean ‘__CPU_SETSIZE’?
   return CPU_SETSIZE;
          ^~~~~~~~~~~
          __CPU_SETSIZE
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1527:10: note: each undeclared identifier is reported only once for each function it appears in
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv_available_parallelism’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1754:12: warning: implicit declaration of function ‘sched_getaffinity’; did you mean ‘SYS_sched_getaffinity’? [-Wimplicit-function-declaration]
   if (0 == sched_getaffinity(0, sizeof(set), &set))
            ^~~~~~~~~~~~~~~~~
            SYS_sched_getaffinity
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1755:10: warning: implicit declaration of function ‘CPU_COUNT’; did you mean ‘__CPU_COUNT_S’? [-Wimplicit-function-declaration]
     rc = CPU_COUNT(&set);
          ^~~~~~~~~
          __CPU_COUNT_S
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c: In function ‘uv_cpumask_size’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/core.c:1531:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
[10/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/timer.c.o -MF CMakeFiles/uv_a.dir/src/timer.c.o.d -o CMakeFiles/uv_a.dir/src/timer.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/timer.c
[11/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/getaddrinfo.c.o -MF CMakeFiles/uv_a.dir/src/unix/getaddrinfo.c.o.d -o CMakeFiles/uv_a.dir/src/unix/getaddrinfo.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/getaddrinfo.c
[12/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/threadpool.c.o -MF CMakeFiles/uv_a.dir/src/threadpool.c.o.d -o CMakeFiles/uv_a.dir/src/threadpool.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/threadpool.c
[13/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/async.c.o -MF CMakeFiles/uv_a.dir/src/unix/async.c.o.d -o CMakeFiles/uv_a.dir/src/unix/async.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/async.c
[14/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/inet.c.o -MF CMakeFiles/uv_a.dir/src/inet.c.o.d -o CMakeFiles/uv_a.dir/src/inet.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/inet.c
[15/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/fs-poll.c.o -MF CMakeFiles/uv_a.dir/src/fs-poll.c.o.d -o CMakeFiles/uv_a.dir/src/fs-poll.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/fs-poll.c
[16/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/getnameinfo.c.o -MF CMakeFiles/uv_a.dir/src/unix/getnameinfo.c.o.d -o CMakeFiles/uv_a.dir/src/unix/getnameinfo.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/getnameinfo.c
[17/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/uv-common.c.o -MF CMakeFiles/uv_a.dir/src/uv-common.c.o.d -o CMakeFiles/uv_a.dir/src/uv-common.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/uv-common.c
[18/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/random-devurandom.c.o -MF CMakeFiles/uv_a.dir/src/unix/random-devurandom.c.o.d -o CMakeFiles/uv_a.dir/src/unix/random-devurandom.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/random-devurandom.c
[19/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/poll.c.o -MF CMakeFiles/uv_a.dir/src/unix/poll.c.o.d -o CMakeFiles/uv_a.dir/src/unix/poll.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/poll.c
[20/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/loop-watcher.c.o -MF CMakeFiles/uv_a.dir/src/unix/loop-watcher.c.o.d -o CMakeFiles/uv_a.dir/src/unix/loop-watcher.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/loop-watcher.c
[21/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/pipe.c.o -MF CMakeFiles/uv_a.dir/src/unix/pipe.c.o.d -o CMakeFiles/uv_a.dir/src/unix/pipe.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/pipe.c
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/pipe.c: In function ‘uv_pipe’:
/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/pipe.c:463:7: warning: implicit declaration of function ‘pipe2’; did you mean ‘pipe’? [-Wimplicit-function-declaration]
   if (pipe2(temp, flags))
       ^~~~~
       pipe
[22/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/loop.c.o -MF CMakeFiles/uv_a.dir/src/unix/loop.c.o.d -o CMakeFiles/uv_a.dir/src/unix/loop.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/loop.c
[23/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/process.c.o -MF CMakeFiles/uv_a.dir/src/unix/process.c.o.d -o CMakeFiles/uv_a.dir/src/unix/process.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/process.c
[24/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/signal.c.o -MF CMakeFiles/uv_a.dir/src/unix/signal.c.o.d -o CMakeFiles/uv_a.dir/src/unix/signal.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/signal.c
[25/38] /usr/bin/cc -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/include -I/usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src -fPIC -g -std=gnu90 -fvisibility=hidden -Wstrict-prototypes -Wextra -Wall -Wno-unused-parameter -fno-strict-aliasing -MD -MT CMakeFiles/uv_a.dir/src/unix/fs.c.o -MF CMakeFiles/uv_a.dir/src/unix/fs.c.o.d -o CMakeFiles/uv_a.dir/src/unix/fs.c.o -c /usr/local/vcpkg/buildtrees/libuv/src/v1.46.0-608af35d5c.clean/src/unix/fs.c
ninja: build stopped: subcommand failed.
```

Reply:
I removed the CI matrix iteration for FreeBSD and retained the CMake preset because it still seems to be necessary for the FreeBSD  native build to succeed.